test/neutrino-test.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+const Network = require('../lib/protocol/network');
+const FullNode = require('../lib/node/fullnode');
+const NeutrinoNode = require('../lib/node/neutrino');
+const {forValue} = require('./util/common');
+const assert = require('bsert');
+
+const network = Network.get('regtest');
+
+describe('Neutrino', function () {
+  this.timeout(10000);
+
+  const fullNode = new FullNode({
+    network: 'regtest',
+    memory: true,
+    listen: true,
+    indexFilter: true,
+    bip157: true
+  });
+
+  async function mineBlocks(n) {
+    while (n) {
+      const block = await fullNode.miner.mineBlock();
+      await fullNode.chain.add(block);
+      n--;
+    }
+  }
+
+  before(async () => {
+    await fullNode.open();
+    await fullNode.connect();
+    await mineBlocks(200);
+  });
+
+  after(async () => {
+    await fullNode.close();
+  });
+
+  describe('No Checkpoints', function () {
+    const neutrinoNode = new NeutrinoNode({
+      network: 'regtest',
+      memory: true,
+      port: 10000,
+      httpPort: 20000,
+      logConsole: true,
+      logLevel: 'debug',
+      neutrino: true,
+      only: '127.0.0.1'
+    });
+
+    before(async () => {
+      await neutrinoNode.open();
+      await neutrinoNode.connect();
+      assert.strictEqual(neutrinoNode.chain.height, 0);
+      assert(neutrinoNode.chain.synced);
+    });
+
+    after(async () => {
+      await neutrinoNode.close();
+    });
+
+    it('should initial sync', async () => {
+      neutrinoNode.startSync();
+      await forValue(neutrinoNode.chain, 'height', fullNode.chain.height);
+    });
+
+    it('should get new blocks headers-only', async () => {
+      await mineBlocks(10);
+      await forValue(neutrinoNode.chain, 'height', fullNode.chain.height);
+      assert.equal(neutrinoNode.chain.height, fullNode.chain.height);
+    });
+
+    it('should cfheaders and getcfilters', async () => {
```

Comment:
I think this entire test should be copied and run twice: after initial sync on L67, and then a second time after the 10 new blocks where it is now.